x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/relevance/curations/CurationsService.java

```diff
@@ -0,0 +1,81 @@
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Manage relevance settings, retrieving and updating the corresponding documents in .ent-search index
+ */
+public class CurationsService {
+
+    public static final String ENT_SEARCH_INDEX = ".ent-search";
+    public static final String CURATIONS_SETTINGS_PREFIX = "curations-";
+    private static final String CONDITIONS_FIELD = "conditions";
+    private static final String PINNED_DOCS_FIELD = "pinned_document_ids";
+    private static final String EXCLUDED_DOCS_IDS = "excluded_document_ids";
+    private static final String CONTEXT_ATTR = "context";
+    private static final String VALUE_ATTR = "value";
+    private static final String ID_ATTR = "_id";
+    private static final String INDEX_ATTR = "_index";
+
+    private final Client client;
+
+    public CurationsService(final Client client) {
+        this.client = client;
+    }
+
+    public CurationSettings getCurationsSettings(String curationId) throws CurationsSettingsNotFoundException {
+        // TODO cache relevance settings, including cache invalidation
+        final Map<String, Object> settingsContent = client.prepareGet(ENT_SEARCH_INDEX, CURATIONS_SETTINGS_PREFIX + curationId)
+            .get()
+            .getSource();
+
+        if (settingsContent == null) {
+            throw new CurationsSettingsNotFoundException("Curation settings " + curationId + " not found");
+        }
+
+        return parseRCurationSettings(settingsContent);
+    }
+
+    private CurationSettings parseRCurationSettings(Map<String, Object> source) throws IllegalArgumentException {
+        // TODO Probably worth to take a look into document mappers in case they can be used for parsing
+        // see org/elasticsearch/index/mapper/DocumentParser.java
+
+        @SuppressWarnings("unchecked")
+        final List<Map<String, Object>> sourceConditions = (List<Map<String, Object>>) source.get(CONDITIONS_FIELD);
+        List<Condition> conditions = sourceConditions.stream().map(this::parseCondition).toList();
+
+        @SuppressWarnings("unchecked")
+        final List<Map<String, Object>> sourcePinnedDocs = (List<Map<String, Object>>) source.get(PINNED_DOCS_FIELD);
+        List<CurationSettings.DocumentReference> pinnedDocs = sourcePinnedDocs.stream().map(this::parseDocumentReference).toList();
+
+        @SuppressWarnings("unchecked")
+        final List<Map<String, Object>> sourceHiddenDocs = (List<Map<String, Object>>) source.get(EXCLUDED_DOCS_IDS);
+        List<CurationSettings.DocumentReference> hiddenDocs = sourceHiddenDocs.stream().map(this::parseDocumentReference).toList();
+
+        return new CurationSettings(pinnedDocs, hiddenDocs, conditions);
+    }
+
+    private CurationSettings.DocumentReference parseDocumentReference(Map<String, Object> sourcePinnedDoc) {
+        return new CurationSettings.DocumentReference((String) sourcePinnedDoc.get(ID_ATTR), (String) sourcePinnedDoc.get(INDEX_ATTR));
```

Comment:
```suggestion
        return new CurationSettings.DocumentReference((String) sourceDocument.get(ID_ATTR), (String) sourceDocument.get(INDEX_ATTR));
```

Including this for convenience if you decide to use my suggested name in the constructor.